onnx/shape_inference/shape_inference.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include "onnx/proto_utils.h"
+#include "onnx/defs/schema.h"
+
+namespace ONNX_NAMESPACE { namespace shape_inference {
+
+struct InferenceContextImpl : public InferenceContext {
```

Comment:
Why do we have this `Impl` instead of just having `InferenceContext`?

Reply:
InferenceContext is just an interface - we need some concrete implementation of it